src/triage/component/catwalk/evaluation.py

```diff
@@ -110,72 +111,177 @@ def _validate_metrics(
+            threshold_value (int) the numeric threshold,
+            parameter_combination (dict) The non-threshold parameter keys and values used
+
+        Returns: (string) A short, human-parseable string
+        """
+        full_params = parameter_combination.copy()
+        if not (threshold_unit == 'percentile' and threshold_value == 100):
+            short_threshold_unit = 'pct' if threshold_unit == 'percentile' else 'abs'
+            full_params[short_threshold_unit] = threshold_value
+        parameter_string = '/'.join([
+            '{}_{}'.format(val, key)
+            for key, val in full_params.items()
+        ])
+        return parameter_string
+
+    def _filter_nan_labels(self, predicted_classes, labels):
+        """Filter missing labels and their corresponding predictions
+
+        Args:
+            predicted_classes (list) Predicted binary classes, of same length as labels
+            labels (list) Labels, maybe containing NaNs
+
+        Returns: (tuple) Copies of the input lists, with NaN labels removed
+        """
+        labels = numpy.array(labels)
+        predicted_classes = numpy.array(predicted_classes)
+        nan_mask = numpy.isfinite(labels)
+        return (
+            (predicted_classes[nan_mask]).tolist(),
+            (labels[nan_mask]).tolist()
+        )
+
+    def _evaluations_for_threshold(
         self,
         metrics,
         parameters,
-        threshold_config,
         predictions_proba,
-        predictions_binary,
         labels,
+        threshold_unit,
+        threshold_value,
     ):
-        """Generate evaluations based on config and create ORM objects to hold them
+        """Generate evaluations for a given threshold in a metric group,
+        and create ORM objects to hold them
 
         Args:
             metrics (list) names of metric to compute
             parameters (list) dicts holding parameters to pass to metrics
-            threshold_config (dict) Unit type and value referring to how any
-                thresholds were computed. Combined with parameter string
-                to make a unique identifier for the parameter in the database
+            threshold_unit (string) the type of threshold, either 'percentile' or 'top_n'
+            threshold_value (int) the numeric threshold,
             predictions_proba (list) Probability predictions
-            predictions_binary (list) Binary predictions
-            labels (list) True labels
+            labels (list) True labels (may have NaNs)
 
         Returns: (list) results_schema.Evaluation objects
         Raises: UnknownMetricError if a given metric is not present in
             self.available_metrics
         """
+
+        # using threshold configuration, convert probabilities to predicted classes
+        predicted_classes = generate_binary_at_x(
+            predictions_proba,
+            threshold_value,
+            unit=threshold_unit
+        )
+        # filter out null labels
```

Comment:
Should we filter out nulls before we subset, so that we get a true amount that we seek?

Reply:
After reading the algorithm page, I see that this is purposefully done.